src/modeling_library/switch/regenerate.jl

```diff
@@ -53,8 +53,8 @@ function regenerate(trace::SwitchTrace{T},
                     argdiffs::Tuple,
                     selection::Selection) where T
     gen_fn = trace.gen_fn
-    index, index_argdiff = args[1], argdiffs[1]
+    index, index_argdiff = trace.index, NoChange()
```

Comment:
The user is allowed to change the branch -- so this is semantically incorrect.

Reply:
The correct fix is to expand `get_args` with `(trace.index, args...)` and then fix any bugs which occur from that.